src/lib.rs

```diff
@@ -24,6 +26,10 @@ pub struct Report {
     pub footer: Option<ReportFooter>,
 }
 
+pub trait Reportable {
+    fn to_reports(&self) -> Vec<Report>;
+}
```

Comment:
```suggestion
pub trait Reportable {
    fn to_reports(&self) -> Vec<&Report>;
}
```

we don't need to clone the report here